### ISSUE TITLE
`SA1010` should not fire for collection expressions

### ISSUE THREAD

**Issue by RenderMichael** [DotNetAnalyzers/StyleCopAnalyzers#3815]
## Description

The following code fires [SA1010OpeningSquareBracketsMustBeSpacedCorrectly](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1010.md)

``` csharp
string[] actual = ["one", "two", "three"];
```

## Expected Behavior

The diagnostic should not fire.

## Actual Behavior

The analyzer fires the diagnostic

**Comment by manfred-brands** [comment]
Duplicate of #3503

@RenderMichael This was fixed in version 1.2.0-beta.556.
